src/transformStateWithClones.js

```diff
@@ -6,8 +6,44 @@
  *
  * @return {Object[]}
  */
+function copyStateAndPush(Obj, arr) {
+  const copyObj = { ...Obj };
+
+  arr.push(copyObj);
+}
+
 function transformStateWithClones(state, actions) {
-  // write code here
+  const historyArray = [];
+  let copyState = { ...state };
+
+  for (const act in actions) {
```

Comment:
Could also use a `for...of` here since `act` is always only used to access `actions`